bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/WakeOnLanUtility.java

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.lgwebos.internal;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.InetAddress;
+import java.net.InterfaceAddress;
+import java.net.NetworkInterface;
+import java.util.Enumeration;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class with utility functions to support Wake On Lan (WOL)
+ *
+ * @author Arjan Mels - Initial contribution
+ * @author Sebastian Prehn - Modification to getMACAddress
+ *
+ */
+@NonNullByDefault
+public class WakeOnLanUtility {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(WakeOnLanUtility.class);
+    private static final Pattern MAC_REGEX = Pattern.compile("(([0-9a-fA-F]{2}[:-]){5}[0-9a-fA-F]{2})");
+
+    private static String COMMAND;
+    static {
+        String os = System.getProperty("os.name").toLowerCase();
+        LOGGER.debug("os: {}", os);
+        if ((os.indexOf("win") >= 0)) {
+            COMMAND = "arp -a %s";
+        } else if ((os.indexOf("mac") >= 0)) {
+            COMMAND = "arp %s";
+        } else { // linux, docker
+            COMMAND = "arping -r -c 1 -C 1 %s";
+        }
+    }
+
+    /**
+     * Get MAC address for host
+     *
+     * @param hostName Host Name (or IP address) of host to retrieve MAC address for
+     * @return MAC address
+     */
+    public static @Nullable String getMACAddress(String hostName) {
+        String cmd = String.format(COMMAND, hostName);
+        LOGGER.debug("cmd: {}", cmd);
+
+        try {
+            Process proc = Runtime.getRuntime().exec(cmd);
```

Comment:
I suggest using org.eclipse.smarthome.io.net.exec.ExecUtil instead since it would simplify this method a bit while making this command line call a bit more robust

Reply:
Right now have to pause the work. No access to the TV for a while due to Corona precautions. Schools closed in Germany, had to move temporarily

Reply:
Np, stay safe.

Reply:
change implemented, but unable to test myself